llvm/include/llvm/Support/MathExtras.h

```diff
@@ -488,13 +488,21 @@ constexpr uint64_t alignTo(uint64_t Value, uint64_t Align) {
   return CeilDiv * Align;
 }
 
+/// Will overflow only if result is not representable in T.
+template <typename U, typename V, typename T = common_uint<U, V>>
+constexpr T alignToPowerOf2(U Value, V Align) {
+  assert(Align != 0 && (Align & (Align - 1)) == 0 &&
+         "Align must be a power of 2");
+  T CeilDiv = divideCeil(Value, Align);
+  return CeilDiv * Align;
+}
+
+/// Fallback when arguments aren't integral.
 constexpr uint64_t alignToPowerOf2(uint64_t Value, uint64_t Align) {
   assert(Align != 0 && (Align & (Align - 1)) == 0 &&
          "Align must be a power of 2");
-  // Replace unary minus to avoid compilation error on Windows:
-  // "unary minus operator applied to unsigned type, result still unsigned"
-  uint64_t NegAlign = (~Align) + 1;
-  return (Value + Align - 1) & NegAlign;
```

Comment:
What was wrong with the old code?

Reply:
I think: `Align - 1` cannot overflow and `Value + (Align - 1)` can only overflow if the result is going to overflow to 0 anyway. Unless you have a counterexample?

Reply:
If Align is 1 it does not and should not overflow. If Align is 2 it should overflow to 0. I still don't see what is wrong with the old code.

Reply:
Sorry about taking so long to debug this, but I finally have definitive proof:

Consider Value=8589934592, Align=1, U = unsigned int, V = unsigned long. Then,

T is computed as unsigned long.
NegAlign is computed as 4294967295 (unsigned long).
Value + (Align - 1) is computed as 8589934592 (unsigned long).
(Value + (Align - 1)) & NegAlign is computed as 0 (unsigned long).

Reply:
I assume this is with long being 64 bits and int being 32 bits? This is just a bug in the way you're calculating NegAlign. You need to extend Align to type T before negating, e.g. with `T NegAlign = (T)0 - Align;`

Reply:
Thanks, fixed now.